### ISSUE TITLE
Bug: [no-shadow] Property functions don't report shadowing on parameter names

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6125]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=MYewdgzgLgBAhgJwOYwLwwAwCgsEsxQCmCAZnMITACqHQwDeWMMAFIkgFwzQL5ICUXKAE8ADoRAl4yANwwA9PJjEEIBExiiAjG2RcefQTBHjJ0pHMXKEq9c1EAmLrs7covMALQA+Y2IlS7JZKYCDWtlgAvkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uQAsBDACYcA7uiiJo0DtEjgwAXxDygA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
const arg = 0

interface Test {
  (arg: string): typeof arg; // error
  p1(arg: string): typeof arg; // error
  p2: (arg: string) => typeof arg; // no error
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-shadow": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

All three `arg: string` parameters should be marked as shadowing `const arg = 0`. This is because the `typeof arg` actually refers to `arg: string`, not `const arg = 0`. Shadowing!

### Actual Result

`p2: (arg: string) => void` _should_ cause an error but _doesn't_, while `(arg: string): void` and `p1(arg: string): void` should and do.

### Additional Info

Reported by @EvanCahill in https://github.com/typescript-eslint/typescript-eslint/issues/6098#issuecomment-1329948334. Though I think the intention might not have been quite what I'm posting here 😄 . _This_ bug report is when `ignoreFunctionTypeParameterNameValueShadow` is _off_.

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.44.0` |
| `@typescript-eslint/parser` | `5.44.0` |
| `TypeScript` | `4.7.2` |
| `ESLint` | `8.15.0` |
| `node` | `web` |

**Comment by EvanCahill** [comment]
All 3 declarations will report a shadowing error when `ignoreFunctionTypeParameterNameValueShadow` is false. `ignoreFunctionTypeParameterNameValueShadow` needs to be explicitly disabled as it defaults to true.

**Comment by JoshuaKGoldberg** [comment]
Ok, thanks - I took a deeper dive and I think I understand. I'll just go ahead and close this as a dup of #6098. Thanks!
